packages/x-components/src/x-modules/search/wiring.ts

```diff
@@ -46,7 +47,7 @@ export const cancelFetchAndSaveSearchResponseWire = wireDispatchWithoutPayload(
  *
  * @public
  */
-export const setOrigin = wireCommit('setOrigin', ({ metadata }) => metadata.origin);
+export const setOrigin = wireCommit('setOrigin', ({ metadata }) => metadata.origin as QueryOrigin);
```

Comment:
We can do this:
```
export const setOrigin = wireCommit('setOrigin', 
  ({ metadata: { feature, location } }) => createQueryOrigin(feature, location)
);
```
or we can create an action to do that, to not overload the wire with logic

Reply:
Done with action